newsletter (#147)
ropensci/roweb3: content/blog/2021-03-15-ropensci-news/index.Rmd

---

```{r setup, include=FALSE}
library("magrittr")
library("rlang")
last_newsletter <- anytime::anytime(params$last_newsletter)
knitr::opts_chunk$set(echo = FALSE)
url <- sprintf(
    "/blog/%s/%s/%s/%s",
    lubridate::year(rmarkdown::yaml_front_matter(knitr::current_input())$date),
    stringr::str_pad(lubridate::month(rmarkdown::yaml_front_matter(knitr::current_input())$date), 2, "0", side = "left"),
    stringr::str_pad(lubridate::day(rmarkdown::yaml_front_matter(knitr::current_input())$date), 2, "0", side = "left"),
    rmarkdown::yaml_front_matter(knitr::current_input())$slug
    )
english <- function(x) {
  as.character(english::english(x))
}
```

Dear rOpenSci friends, it's time for our monthly news roundup!
Some housekeeping first:
We've been changing the infrastructure of our newsletter a bit so please update your RSS and JSON feeds to `https://ropensci.org/tags/newsletter/index.xml` and `https://ropensci.org/tags/newsletter/index.json`, respectively.
You can read this post [on our blog](`r url`).
Now let's dive into the activity at and around rOpenSci!

## rOpenSci HQ

Our R-universe project now has [its own page on our website](/r-universe/)! :rocket:

> Regarding another exciting project of ours... We've run the first comm call of our series on [statistical software review project](/stat-software-review/)! [rOpenSci Statistical Software Testing and Peer Review](/commcalls/feb2021-statsreview/) took place on March the 2d. The video recording (including closed captions) and resources have been posted.

Reviewer: ```suggestion
Regarding another exciting project of ours... We've run the first comm call of our series on the [statistical software review project](/stat-software-review/)! [rOpenSci Statistical Software Testing and Peer Review](/commcalls/feb2021-statsreview/) took place on March 2nd. The video recording (including closed captions) and resources have been [posted](/commcalls/feb2021-statsreview/).
```

new <- citations[length(old):length(citations)]

format_package <- function(package, packages = packages) {
  if (package %in% packages) {
    package <- sprintf("[**%s**](https://docs.ropensci.org/%s)", package, package)
  } else {
    package <- sprintf("**%s**", package)
  }
  package
}

format_one <- function(citation, packages) {
  packages <- toString(purrr::map_chr(citation$name, format_package, packages = packages))
  sprintf("* %s in %s\n\n", packages, citation$citation)
}

cat(
  paste0(
    sort(purrr::map_chr(new, format_one, packages = registry$package)),
    collapse = ""
  )
)

```

Thank you for citing our tools!

## Use cases

```{r usecases}
# rerun get_use_cases.R at the same time
usecases <- jsonlite::read_json("../../../data/usecases/usecases.json")
get_one_case <- function(usecase) {
  tibble::tibble(
    title = usecase$title,
    reporter = usecase$reporter,
    url = usecase$url,
    image = usecase$image,
    date = anytime::anydate(usecase$date)
  )
}
usecases <- purrr::map_df(usecases, get_one_case)
usecases <- dplyr::filter(usecases, date >= anytime::anydate("2021-01-01"))#as.Date(last_newsletter))
usecases <- split(usecases, seq(nrow(usecases)))
```

> `r snakecase::to_sentence_case(english(length(usecases)))` use cases of our packages and resources have been reported since the 1st of January.

Reviewer: Should the "1st of January" be programatically created?

Author: No but I'll need to change this next time. Next time I'll use the date of the last newsletter but as you know last time we had missed a few use cases. :grin:

```{r usecases2, results='asis'}
format_case <- function(usecase) {
  string <- sprintf("* [%s](%s). Reported by %s.", usecase$title, usecase$url, usecase$reporter)
  string
}
cat(
  paste0(
    purrr::map_chr(usecases, format_case),
    collapse = "\n\n"
  )
)
```

Explore [other use cases](/usecases) and [report your own](https://discuss.ropensci.org/c/usecases/10)!

> ## Call For maintainers

Reviewer: ```suggestion
## Call for maintainers
```

Afraid to forget something when you **submit your package to CRAN**?
A check list could help!
You can use [`usethis::use_release_issue()`](https://usethis.r-lib.org/reference/use_release_issue.html), [number one "secret" [of the tidyverse team] for avoiding multiple rounds of review with CRAN](https://twitter.com/hadleywickham/status/1359852335330164737).
Now, if you want to add your own specific bullet e.g. ["Update my brag document"](https://jvns.ca/blog/brag-documents/) or ["Consider writing a tech note for rOpenSci blog"](https://blogguide.ropensci.org/index.html), you can add an [unexported `release_bullets()` function](https://github.com/r-lib/usethis/issues/1183) returning a character vector to your package.

```r
release_bullets <- function() {
  c(
    "Update my brag document.",
    "Contact rOpenSci blog editors to schedule a tech note."
  )
}
```

For inspiration you can search the [CRAN mirror maintained by R-hub](https://github.com/search?q=release_bullets+user%3Acran&type=Code).
For more resources around CRAN, see [Isabella Velásquez's post "You CRAN Do It, What CRAN First-Timers Should Know"](https://ivelasq.rbind.io/blog/you-cran-do-it/).

## Last words

> Thanks for reading! If you want to want to get involved with rOpenSci, check out our [Contributing Guide](https://contributing.ropensci.org) that can help direct you to the right place, whether you want to make code contributions, non-code contributions, or other things like sharing use cases.

Reviewer: ```suggestion
Thanks for reading! If you want to get involved with rOpenSci, check out our [Contributing Guide](https://contributing.ropensci.org) that can help direct you to the right place, whether you want to make code contributions, non-code contributions, or contribute in other ways like sharing use cases.
```
